From 205c305922cdbded91ff8fadbaad7959bdb497bc Mon Sep 17 00:00:00 2001
From: Ian Kent <raven@themaw.net>
Date: Wed, 11 Aug 2010 09:33:09 +0800
Subject: [PATCH 2/2] autofs-5.0.5 - remove ERR_remove_state() openssl call

autofs should never have had to use ERR_remove_state() so remove that call.

Lukas: Resolve a conflict in "CHANGELOG" that occured due to cherry
picking Kerberos related patches from the 5.0.6 development branch.

Conflicts:

	CHANGELOG

Signed-off-by: Lukas Fleischer <archlinux@cryptocrack.de>
---
 CHANGELOG             |    4 ++++
 modules/lookup_ldap.c |   12 +-----------
 2 files changed, 5 insertions(+), 11 deletions(-)

diff --git a/CHANGELOG b/CHANGELOG
index e734cb3..1a20a81 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,7 @@
+??/??/20?? autofs-5.0.6
+-----------------------
+- remove ERR_remove_state() openssl call.
+
 03/09/2009 autofs-5.0.5
 -----------------------
 - fix dumb libxml2 check
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 2ecf5fe..1221c2c 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -168,18 +168,8 @@ int unbind_ldap_connection(unsigned logopt, LDAP *ldap, struct lookup_context *c
 	int rv;
 
 #ifdef WITH_SASL
-	/*
-	 * The OpenSSL library can't handle having its message and error
-	 * string database loaded multiple times and segfaults if the
-	 * TLS environment is not reset at the right times. As there
-	 * is no ldap_stop_tls call in the openldap library we have
-	 * to do the job ourselves, here and in lookup_done when the
-	 * module is closed.
-	 */
-	if (ctxt->use_tls == LDAP_TLS_RELEASE) {
-		ERR_remove_state(0);
+	if (ctxt->use_tls == LDAP_TLS_RELEASE)
 		ctxt->use_tls = LDAP_TLS_INIT;
-	}
 	autofs_sasl_unbind(ctxt);
 #endif
 
-- 
1.7.5.1

